src/status_im2/contexts/onboarding/create_password/view.cljs

```diff
@@ -1,42 +1,206 @@
+        error?        (and show-password-validation?
+                           (not passwords-match?)
+                           (not empty-password?))]
+    [:<>
+     [password-with-hint
+      {:hint           {:text   (i18n/label :t/password-creation-hint)
+                        :status hint-1-status
+                        :shown  true}
+       :placeholder    (i18n/label :t/password-creation-placeholder-1)
+       :on-change-text on-change-password
+       :on-focus       #(on-input-focus :password)}]
+     [rn/view {:style style/space-between-inputs}]
+     [password-with-hint
+      {:hint           {:text   hint-2-text
+                        :status hint-2-status
+                        :shown  (and (not empty-password?)
+                                     show-password-validation?)}
+       :error?         error?
+       :placeholder    (i18n/label :t/password-creation-placeholder-2)
+       :on-change-text on-change-repeat-password
+       :on-focus       #(on-input-focus :repeat-password)
+       :on-blur        on-blur-repeat-password}]]))
+
+(def strength-status
+  {1 :very-weak
+   2 :weak
+   3 :okay
+   4 :strong
+   5 :very-strong})
+
+(defn help
+  [{{:keys [lower-case? upper-case? numbers? symbols?]} :validations
+    password-strength                                   :password-strength}]
+  [rn/view
+   [quo/strength-divider {:type (strength-status password-strength :info)}
+    (i18n/label :t/password-creation-tips-title)]
+   [rn/view {:style style/password-tips}
+    [quo/tips {:completed? lower-case?}
+     (i18n/label :t/password-creation-tips-1)]
+    [quo/tips {:completed? upper-case?}
+     (i18n/label :t/password-creation-tips-2)]
+    [quo/tips {:completed? numbers?}
+     (i18n/label :t/password-creation-tips-3)]
+    [quo/tips {:completed? symbols?}
+     (i18n/label :t/password-creation-tips-4)]]])
+
+(defn password-validations
+  [password]
+  (let [validations (juxt utils.string/has-lower-case?
+                          utils.string/has-upper-case?
+                          utils.string/has-numbers?
+                          utils.string/has-symbols?
+                          #(utils.string/at-least-n-chars? % 10))]
+    (->> password
+         (validations)
+         (zipmap [:lower-case? :upper-case? :numbers? :symbols? :long-enough?]))))
+
+(defn calc-password-strength
+  [validations]
+  (->> (vals validations)
+       (filter true?)
+       (count)))
+
+(defn password-form
+  [{:keys [scroll-to-end-fn]}]
+  (let [state                     (reagent/atom {:password                  ""
+                                                 :repeat-password           ""
+                                                 :accepts-disclaimer?       false
+                                                 :focused-input             nil
+                                                 :show-password-validation? false})
+        password                  (reagent/cursor state [:password])
```

Comment:
I understand the use case for cursors @ulisesmac, but what advantage do they bring in this particular case? Wouldn't it be simpler if you simply removed the `state` ratom and only defined the individual ratoms? Performance wise it would be the same.

Reply:
Totally agree!
I was going to do it using single ratoms, but I've seen this pattern in many places across the codebase and just copied it. I've just changed it to use multiple ratoms